### ISSUE TITLE
[BUG] Attempting to open an invalid file causes AYAB to crash

### ISSUE THREAD

**Issue by jllyh** [AllYarnsAreBeautiful/ayab-desktop#745]
When I attempted to open a file with an .STP extension, AYAB crashed. I believe that it's because the file itself is invalid (meta information about the actual .STP file, perhaps). The 'crash' behavior needs better handling ("This is not a valid file" or somesuch). I would attach an example but the .STP type could not be attached here. 

Computer/OS: Dell Inspiron, Windows 10
Release 1.0.0.-beta4
Knitting machine: N/A
AYAB hardware: N/A

**Comment by dl1com** [comment]
You could add an extension to the file (e.g. .txt) so you might trick Github to allow uploading.
But it seems there has also been discussion in Discord about your issue - is this still relevant?
BR

**Comment by jllyh** [comment]
Yes. The discussion (and file) can be found [here ](https://discord.com/channels/1079442610332110849/1079442748307947541/1323784491344592977)but I think the 'crash' behavior could be reproduced with any file that has a valid extension but is not actually a valid file type.

**Comment by jonathanperret** [comment]
Here's the file uploaded by @jllyh to Discord, renamed to have a `.txt` extension so that GitHub accepts it.

[Invalid.stp.txt](https://github.com/user-attachments/files/18318489/Invalid.stp.txt)
